src/phoenix/trace/langchain/tracer.py

```diff
@@ -234,3 +281,43 @@ def _persist_run(self, run: Run) -> None:
             self._convert_run_to_spans(run.dict())
         except Exception:
             logger.exception("Failed to convert run to spans")
+
+    def on_chat_model_start(
```

Comment:
The way this currently works is that the callback system first tries calling `on_chat_model_start` for chat models. If that hook is not implemented, it falls back to `on_llm_start`. Somewhat confusingly, there is no `on_chat_model_end` hook, only a `on_llm_end` hook that is call regardless of whether `on_chat_model_start` is implemented.

https://github.com/langchain-ai/langchain/blob/f0198354d93e7ba8b615b8fd845223c88ea4ed2b/libs/langchain/langchain/callbacks/manager.py#L381